server/src/Classes/ActionClass.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Gernzy\Server\Classes;
+
+class ActionClass
+{
+    protected $meta;
+    protected $dataModified = [];
+    protected $dataOriginal = [];
+
+    public function __construct($dataOriginal = null)
+    {
+        $this->meta = [];
+        $this->dataModified = [];
+        $this->dataOriginal = $dataOriginal;
+        $this->eventPreventDefault = false;
+    }
+
+    /**
+     * Set the meta of the action object, which is the name of the actions that are configure to fire on the event.
+     */
+    public function setMeta($action)
+    {
+        array_push($this->meta, [
+            'action' => $action
+        ]);
+    }
+
+    /**
+     * This is the main funtion used by third parties to attach data to the Action Place holder object, to be used by next package.
+     * This data will be appended to the $dataModified array property of this object
+     * @param string
+     * @param $var
+     */
+    public function attachData($actionName, $data = [])
+    {
+        array_push($this->dataModified, [
```

Comment:
@AlexeiaW you can usually just use array_merge for "extending" the array, rather than push. Although push will work okay in this scenario as well.

Reply:
@AlexeiaW sorry push makes sense in this context, because you're building a list of data per action, ignore above.